src/pages/TermsOfService.ts

```diff
@@ -0,0 +1,148 @@
+
+  const accept = async () => {
+    try {
+      setBusy(true);
+      const acceptTosResponse = await Ajax().User.acceptTos();
+      const { enabled } = acceptTosResponse || { enabled: false };
+      if (enabled) {
+        const termsOfServiceResponse = await Ajax().User.getTermsOfServiceComplianceStatus();
+        const termsOfService = termsOfServiceResponse || { userHasAcceptedLatestTos: false, permitsSystemUsage: false };
+
+        const registrationStatus = 'registered';
+        authStore.update((state) => ({ ...state, registrationStatus, termsOfService }));
+        Nav.goToPath('root');
+      } else {
+        reportError('Error accepting TOS, unexpected backend error occurred.', new Error('Unexpected backend error'));
+      }
+    } catch (error) {
+      reportError('Error accepting Terms of Service', error);
+    } finally {
+      setBusy(false);
+    }
+  };
+
+  const reject = async () => {
+    try {
+      setBusy(true);
+      await Ajax().User.rejectTos();
+    } catch (error) {
+      reportError('Error rejecting Terms of Service', error);
+    } finally {
+      setBusy(false);
+      signOut('declinedTos');
+    }
+  };
+
+  const continueButton = () => {
+    Nav.goToPath('root');
+  };
+
+  return div(
+    {
+      role: 'main',
+      style: { padding: '1rem', minHeight: '100%', display: 'flex', justifyContent: 'center', alignItems: 'center' },
+    },
+    [
+      img({
+        src: scienceBackground,
+        alt: '',
+        style: { position: 'fixed', top: 0, left: 0, zIndex: -1 },
+      }),
+      div(
+        {
+          style: {
+            backgroundColor: 'white',
+            borderRadius: 5,
+            width: 800,
+            maxHeight: '100%',
+            padding: '2rem',
+            boxShadow: Style.standardShadow,
+          },
+        },
+        [
+          h1({ style: { color: colors.dark(), fontSize: 38, fontWeight: 400 } }, ['Terra Terms of Service']),
+          !acceptedLatestTos &&
+            div({ style: { fontSize: 18, fontWeight: 600 } }, ['Please accept the Terms of Service to continue.']),
+          div(
+            { style: { height: '50vh', overflowY: 'auto', lineHeight: 1.5, marginTop: '1rem', paddingRight: '1rem' } },
+            [
+              !tosText
+                ? centeredSpinner()
+                : h(
+                    MarkdownViewer,
+                    {
+                      renderers: {
+                        link: newWindowLinkRenderer,
+                        heading: (text, level) => `<h${level} style="margin-bottom: 0">${text}</h${level}>`,
+                      },
+                    },
+                    [tosText]
+                  ),
+            ]
+          ),
+          !usageAllowed &&
+            !!tosText &&
+            div({ style: { display: 'flex', justifyContent: 'flex-end', marginTop: '2rem' } }, [
+              h(ButtonSecondary, { style: { marginRight: '1rem' }, onClick: reject }, ['Decline and Sign Out']),
```

Comment:
This is really the only functional change the pr has, adding the onclick: reject, before it was just signout, but now it will do the reject api call to sam and then sign out.

Reply:
Similar to the accept button, should this and the other "Decline and Sign Out" button be disabled while `busy` is true?